incident_viewer_demo/src/LineChart/LineChart.jsx

```diff
@@ -309,88 +309,87 @@ class LineChart extends PureComponent {
         .attr('x2', xScale(closest));
 
       const highlightedDate = moment(closest);
-      const goodYmax = yScale(goodIncidentSeries.download_speed_mbps_median);
-      const badYmax = yScale(badIncidentSeries.download_speed_mbps_median);
+      const goodYmax = yScale(incident.goodPeriodMetric);
+      const badYmax = yScale(incident.badPeriodMetric);
       this.infoHoverBox.selectAll('*').remove();
 
       if (hasIncident) {
-        const width = xScale(goodIncidentSeries.end) - xScale(goodIncidentSeries.start);
+        const width = xScale(incident.goodPeriodEnd) - xScale(incident.goodPeriodStart);
         const rectFitsText = width > 180; // NOTE: This also must be manually tuned. It hides hover text in the case 
                                           // that the area is too small for the text to fit.
         const verticalTextShifter = 0.45; // NOTE: This needs to be tuned based on font size and number of lines :(
 
         // Draw the hover state for the good period information
-        if (highlightedDate.isBefore(goodIncidentSeries.end) && highlightedDate.isSameOrAfter(goodIncidentSeries.start) && mouseY > goodYmax) {
+        if (highlightedDate.isBefore(incident.goodPeriodEnd) && highlightedDate.isSameOrAfter(incident.goodPeriodStart) && mouseY > goodYmax) {
           this.infoHoverBox.append('rect')
           .classed("good-incident-area", true)
-          .attr('x', xScale(goodIncidentSeries.start))
+          .attr('x', xScale(incident.goodPeriodStart))
           .attr('y', goodYmax)
           .attr('width', width)
           .attr('height', plotAreaHeight-goodYmax);
 
           if (rectFitsText) {
```

Comment:
One note about this block and the two similar ones below is that they have been changed in the textFix branch, which reformats the way that hover text is displayed. We should be deliberate about the fact that the textFix merge needs to use this JSON parsing and its own display formatting (ie: get rid of those tspans!).